platform/darwin/src/local_glyph_rasterizer.mm

```diff
@@ -45,47 +46,72 @@ CTFontDescriptor configuration option (although we'd have to override font size
 using CGContextHandle = CFHandle<CGContextRef, CGContextRef, CGContextRelease>;
 using CFStringRefHandle = CFHandle<CFStringRef, CFTypeRef, CFRelease>;
 using CFAttributedStringRefHandle = CFHandle<CFAttributedStringRef, CFTypeRef, CFRelease>;
+using CFMutableArrayRefHandle = CFHandle<CFMutableArrayRef, CFTypeRef, CFRelease>;
 using CFDictionaryRefHandle = CFHandle<CFDictionaryRef, CFTypeRef, CFRelease>;
+using CTFontRefHandle = CFHandle<CTFontRef, CFTypeRef, CFRelease>;
 using CTFontDescriptorRefHandle = CFHandle<CTFontDescriptorRef, CFTypeRef, CFRelease>;
 using CTLineRefHandle = CFHandle<CTLineRef, CFTypeRef, CFRelease>;
 
 class LocalGlyphRasterizer::Impl {
 public:
     Impl(const optional<std::string> fontFamily_)
-        : fontFamily(fontFamily_)
-        , fontHandle(NULL)
-    {}
+    {
+        fallbackFontNames = [[NSUserDefaults standardUserDefaults] stringArrayForKey:@"MGLIdeographicFontFamilyName"];
+        if (fontFamily_) {
+            fallbackFontNames = [fallbackFontNames ?: @[] arrayByAddingObjectsFromArray:[@(fontFamily_->c_str()) componentsSeparatedByString:@"\n"]];
```

Comment:
The primary thing I’m looking for feedback on is what to do about this single-string argument to the LocalGlyphRasterizer constructor. My understanding is that other platforms like Android need to select and validate a single font at the SDK level, whereas iOS/macOS has outgrown that approach and needs to maintain a whole list of fonts that can only be validated by the system on the fly.

Should we change the signature of this constructor across all platforms to allow for a `std::vector` of font names? Alternatively, would it be OK for the iOS/macOS implementation to ignore the single-string name or treat it as just one more fallback (as currently implemented in this PR)?

Reply:
I prefer the idea of using `std::vector` here. Though could we land this PR as is and address that later?

Reply:
Perhaps – it just feels wrong to smuggle a newline-delimited list through a single-string API to split it apart on the other end.

Reply:
> The primary thing I’m looking for feedback on is what to do about this single-string argument to the LocalGlyphRasterizer constructor.

> Should we change the signature of this constructor across all platforms to allow for a std::vector of font names? Alternatively, would it be OK for the iOS/macOS implementation to ignore the single-string name or treat it as just one more fallback (as currently implemented in this PR)?

@alexshalamov I think you're the most familiar with this part of the code. Is there any reason GL Native can't accept a whole list of fonts here? 

If we agree to make this change at the core level, I also think that it would be better in a separate PR.

cc @tmpsantos @tobrun @knov @zugaldia

Reply:
I had been trying to minimize changes to the cross-platform interface, but this constructor is only called from platform-specific code, so the cross-platform code in mbgl doesn’t particularly care about the signature. If folks are OK with a little additional churn, it would be great to change this method to take a `std::optional<std::vector<std::string>>`, with `nullptr` disabling local glyph rasterization and an empty vector using the system’s default fallback.